yo/api_server.py

```diff
@@ -101,10 +100,15 @@ class YoAPIServer(YoBaseService):
            list: list of notifications updated
        """
        retval = []
-       if test and self.testing_allowed:
-          for notify_id in ids:
-              self.test_notifications.mark_notification_unseen(notify_id)
-              retval.append(self.test_notifications.get_notification(notify_id))
+       if test:
```

Comment:
I've explained this before in issue #50 but there's a reason this is here - it's so that we can test external apps and so that we can compare behaviour between the mock API and real data.

The testing_allowed flag can be disabled in config, and for production it should be disabled by default. If we ever open up parts of the API to third parties, we should replace the static flag with a check against the client credentials to see if they have access to the testing API.

Removing the testing API will only slow down development, leaving it in place doesn't harm anything and there's a reason it's in there.

Reply:
> it's so that we can test external apps

External apps are responsible for their own testing

> so that we can compare behaviour between the mock API and real data

The api behavior contract should be defined as a schema, which can be used by us, external apps, or anyone else to assert api contract compliance

> Removing the testing API will only slow down development

Thats why I didn't reject this and haven't removed the code yet, but all testing logic will be moved to testing after mvp

Reply:
"External apps" in this context includes stuff like condenser, if we're talking about third-party apps then i'd say it's useful to have the test API available anyway, pretty much any serious webservice offers a testing mode.

The behaviour contract is currently being defined as a schema, but having running code just makes things easier.

In general, I strongly believe that we should keep the mock stuff in place, we can do the same stuff inside the test suite of course but it's so much simpler to have support in the main codebase even if it's stylistically incorrect - it's pragmatic and useful.